draft-normington-jsonpath-latest.md

```diff
@@ -84,43 +108,266 @@ informative:
+
+# JSONPath Examples
+
+This section provides some more examples for JSONPath expressions.
+The examples are based on a simple JSON data item patterned after a
+typical XML example representing a bookstore (that also has bicycles):
+
+~~~~json
+{ "store": {
+    "book": [ 
+      { "category": "reference",
+        "author": "Nigel Rees",
+        "title": "Sayings of the Century",
+        "price": 8.95
+      },
+      { "category": "fiction",
+        "author": "Evelyn Waugh",
+        "title": "Sword of Honour",
+        "price": 12.99
+      },
+      { "category": "fiction",
+        "author": "Herman Melville",
+        "title": "Moby Dick",
+        "isbn": "0-553-21311-3",
+        "price": 8.99
+      },
+      { "category": "fiction",
+        "author": "J. R. R. Tolkien",
+        "title": "The Lord of the Rings",
+        "isbn": "0-395-19395-8",
+        "price": 22.99
+      }
+    ],
+    "bicycle": {
+      "color": "red",
+      "price": 19.95
+    }
+  }
+}
+~~~~
+{: fig-example-item title="Example JSON data item"}
+
+The examples in {{tbl-example}} presume an underlying script language
+that allows obtaining the number of items in an array, testing for the
+presence of a map member, and performing numeric comparisons of map
+member values with a constant.
+
+| XPath                  | JSONPath                                  | Result                                                       |
+|------------------------|-------------------------------------------|--------------------------------------------------------------|
+| `/store/book/author`   | `$.store.book[*].author`                  | the authors of all books in the store                        |
+| `//author`             | `$..author`                               | all authors                                                  |
+| `/store/*`             | `$.store.*`                               | all things in store, which are some books and a red bicycle. |
+| `/store//price`        | `$.store..price`                          | the price of everything in the store.                        |
+| `//book[3]`            | `$..book[2]`                              | the third book                                               |
+| `//book[last()]`       | `$..book[(@.length-1)]`<br>`$..book[-1:]` | the last book in order.                                      |
+| `//book[position()<3]` | `$..book[0,1]`<br>`$..book[:2]`           | the first two books                                          |
+| `//book[isbn]`         | `$..book[?(@.isbn)]`                      | filter all books with isbn number                            |
+| `//book[price<10]`     | `$..book[?(@.price<10)]`                  | filter all books cheapier than 10                            |
+| `//*`                  | `$..*`                                    | all Elements in XML document. All members of JSON data item. |
```

Comment:
I'm not sure that there is much value in keeping the first column given that JSONPath is already established.

Please can we make the result descriptions consistently end or not end with a period. I prefer to avoid the period.

Reply:
I think the first column is helpful for those familiar with XPath, and helps to coalesce the concepts from the preceding table.